### ISSUE TITLE
[3.3.5] Quest: Free at Last (12099)

### ISSUE THREAD

**Issue by Treeston** [TrinityCore/TrinityCore#16577]
3.3.5 8e6fb3b1c50f

[Quest: Free at Last](http://www.wowhead.com/quest=12099/free-at-last)
No credit is awarded when using the [Gavrock's Runebreaker](http://www.wowhead.com/item=36796/gavrocks-runebreaker) on a [Runed Giant](http://www.wowhead.com/npc=26417/runed-giant).

**Comment by Saben65** [comment]
Suggested fix
```SQL
DELETE FROM `smart_scripts` WHERE `entryorguid`=26417 AND `id`=3;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(26417,0,3,0,8,0,100,0,47604,0,0,0,33,24329,0,0,0,0,0,7,0,0,0,0,0,0,0,"Runed Giant - On Spellhit 'Gavrock's Runebreaker' - Quest Credit ''");
```

**Comment by dr-j** [comment]
Read the comments http://www.wowhead.com/quest=12099/free-at-last#comments

More than one person saying its not 100% chance to work and it currently does work but its 25% chance but as players say there if it fails you can use again on the same giant.

**Comment by Treeston** [comment]
Correct. However, even when the giant flags friendly, it still doesn't award credit.
There's not a 100% chance for the giant to flag friendly (this is correct), but it should award credit _if_ it flags friendly (this doesn't happen).

**Comment by dr-j** [comment]
ah try this

~~~sql
    UPDATE `smart_scripts` SET `action_type`=45, `action_param1`=1, `action_param2`=1, `target_type`=1, `comment`='Runed Giant - On Script - Set Data Self' WHERE  `entryorguid`=2641700 AND `source_type`=9 AND `id`=2 AND `link`=0;

    DELETE FROM `smart_scripts` WHERE `entryorguid`=26417 AND `source_type`=0;
    INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
    (26417, 0, 0, 0, 11, 0, 100, 0, 0, 0, 0, 0, 11, 47329, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Runed Giant - On Spellhit \'Gavrock\'s Runebreaker\' - Run Script (No Repeat)'),
    (26417, 0, 1, 0, 38, 0, 100, 0, 1, 1, 0, 0, 33, 26783, 0, 0, 0, 0, 0, 12, 1, 0, 0, 0, 0, 0, 0, 'Runed Giant - On Script - Quest Credit \'Free at Last\''),
    (26417, 0, 2, 3, 8, 0, 25, 0, 47604, 0, 10000, 10000, 64, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Runed Giant - On Spell Hit - Store Target List'),
    (26417, 0, 3, 0, 61, 0, 0, 0, 0, 0, 10000, 10000, 80, 2641700, 2, 2, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Runed Giant - On Spell Hit - Start Script for Free At Last');

**Comment by Saben65** [comment]
Haven't tried @dr-j fix, but this works as well.

```SQL
DELETE FROM `smart_scripts` WHERE `entryorguid`=26417 AND `id`=3;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(26417,0,3,0,61,0,100,0,47604,0,10000,10000,33,24329,0,0,0,0,0,7,0,0,0,0,0,0,0,"Runed Giant - On Spellhit 'Gavrock's Runebreaker' - Quest Credit ''");
```

I just linked the quest credit to the script where it becomes friendly.

**Comment by Treeston** [comment]
Right, so. This quest again, and without reliable repro I'm afraid. Something is just off.

I'm only getting a roughly 5% success rate of the Runebreaker on giants, when I (anecdotally, admittedly) remember it being more like 25%.

Could someone double-check the probability numbers just to be sure? Could be RNG, of course, but I've accrued a pretty decent sample size at this point.

**Comment by Killyana** [comment]
It's 25% to run the action list 2641700.
``` sql
entryorguid, source_type, id, link, event_type, event_phase_mask, event_chance, event_flags, event_param1, event_param2, event_param3, event_param4, action_type, action_param1, action_param2, action_param3, action_param4, action_param5, action_param6, target_type, target_param1, target_param2, target_param3, target_x, target_y, target_z, target_o, comment
 26417, 0, 3, 0, 61, 0, 100, 0, 0, 0, 10000, 10000, 87, 2641700, 2641701, 2641701, 2641701, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, Runed Giant - On Spell Hit - Start Script for Free At Last

**Comment by Treeston** [comment]
Are we sure the 25% chance isn't additionally handled somewhere else (which would result in a 7.5% chance?)

**Comment by SnapperRy** [comment]
For a quick 'n' dirt test, try setting action_param2 to 2641700. It should result in a 50% chance to get a successful attempt.

**Comment by dr-j** [comment]
Actually there is nothing wrong with credit but there is bug here basically as it stands player gets one chance per giant to "free it" which is 25% chance if you fail to free it it turns to weakened giant (non elite) when I originally scripted this it worked so that if it failed the item can be used again on the same giant until sucessful.

But recently for some reason this no longer works but cant find problem in sai.

Basically quest item will not work at all on giants once they are weakened but this worked until recently and not sure what changed.

These change entry from runed giant to weakened giant.

**Comment by Aokromes** [comment]
this issue is still valid?
